### PR TITLE
Support BaguaStrategy in LightningLite

### LINKED ISSUE #12500: Support BaguaStrategy in LightningLite
## 🚀 Feature

Add Bagua to the supported strategy types in Lite. 

### Motivation

Users who want to try Bagua but are hesitant to convert fully to Lightning can take a shortcut and just use Lite to benchark their code against SOTA strategies with only a few code changes. 

### Pitch

In theory Lite is strategy agnostic and can support Bagua out of the box. However, we should verify that first with a simple test and then add it to the supported types. 

### Alternatives

No.

### Additional context

#12069

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #12500 
Prep PR: #12692 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/lite/lite.py [hunk 1]
```diff
@@ -410,7 +410,13 @@ def _run_impl(self, run_method: Callable, *args: Any, **kwargs: Any) -> Any:
     def _run_with_strategy_setup(self, run_method: Callable, *args: Any, **kwargs: Any) -> Any:
         self._strategy.setup_environment()
         with self._strategy.model_sharded_context(), _replace_dataloader_init_method():
-            return run_method(*args, **kwargs)
+            result = run_method(*args, **kwargs)
+
+        # TODO: Capture this inside a teardown
+        if isinstance(self._strategy, BaguaStrategy) and self._strategy._bagua_algorithm == "async":
+            self._strategy.model.module.bagua_algorithm.abort(self._strategy.model.module)
```
Comment (awaelchli) [inline pytorch_lightning/lite/lite.py:417]: This is required for the async method. Does that mean we would need a self.teardown() method that the user calls inside Lite? Or do we need to drop support for async in Lite?
